### PR TITLE
[16.0][IMP] ir_actions_report: attachments creation maintainability improvement

### PR DESCRIPTION
**Pull request by yankinmax**
Description of the issue/feature this PR addresses:

Previously, an `ir_actions_report` refactor landed on:
- [ ] https://github.com/odoo/odoo/pull/85150

It totally makes sense to create `ir.attachment` from the streams if needed during PDF generation.
For some of the projects we still need to override attachment values.
This PR provides a hook to prepare attachment values needed for attachments creation during the pdf report generation.
Nothing changes in the sense of business logic.
Ping @smetl @xmo-odoo 

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by yankinmax** [comment]
CLA agreement issue will be solved after this PR will be integrated to odoo source code:
- [ ] https://github.com/odoo/odoo/pull/137603

File: odoo/addons/base/models/ir_actions_report.py [hunk 1]
```diff
@@ -198,6 +198,44 @@ def retrieve_attachment(self, record):
                 ('res_id', '=', record.id)
         ], limit=1)
 
+    def _prepare_pdf_report_attachment_vals_list(self, report, streams):
```
Comment (ivantodorovich) [inline odoo/addons/base/models/ir_actions_report.py:201]: maybe keep this hook method closer to the upper method.
This is line 200ish and `_render_qweb_pdf` is on line 800ish :)

File: odoo/addons/base/models/ir_actions_report.py [hunk 2]
```diff
@@ -804,6 +804,44 @@ def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
             collected_streams[False] = {'stream': pdf_content_stream, 'attachment': None}
 
         return collected_streams
+
```
Comment (ivantodorovich) [inline odoo/addons/base/models/ir_actions_report.py:807]: There seems to be a linting error here (CI/style)



> odoo/odoo/addons/base/models/ir_actions_report.py:807C0303(trailing-whitespace) coll 0: Trailing whitespace
>
> [odoo/odoo/addons/base/models/ir_actions_report.py:807](https://github.com/odoo/odoo/blob/03ad3eb5e3afdc85804437b0d37c72a0c055a248/odoo/addons/base/models/ir_actions_report.py#L807)
> 
> C0303(trailing-whitespace) coll 0: Trailing whitespace

**Comment by ivantodorovich** [comment]
Hi @smetl 👋🏻 as you worked on the mentioned PR, you might be interested by it
Thanks in advance ^^

**Review by smetl — APPROVED** [review]

**Review by vvrossem — COMMENTED** [review]
LGTM. Thank you for this improvement @yankinmax :pray:

**Comment by vvrossem** [comment]
Hello @xmo-odoo, @rco-odoo, 
ci/security is failing for this PR: an issue with `safe_eval`.
However, this PR externalizes in a method the bit where `safe_eval` was initially used.
What should we do about it? 
Thank you for your insights :pray:

**Comment by yankinmax** [comment]
thank you @xmo-odoo for taking time to review this PR. CLA is taken into account now.

**Comment by smetl** [comment]
@robodoo r+

**Pull request closed without merging**